### PR TITLE
Remove deprecated frame processor

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- remove deprecated frame processor infrastructure and obsolete builder methods
- adopt length-delimited codec across tests and helpers
- document codec-based framing and drop processor docs

## Testing
- `make fmt`
- `make lint`
- `make test`
- `make markdownlint`

------
https://chatgpt.com/codex/tasks/task_e_68b0925ebe48832299603ca21dc5c28d

## Summary by Sourcery

Remove the deprecated FrameProcessor infrastructure in favor of using tokio_util's LengthDelimitedCodec end-to-end, enrich length-prefix handling, and clean up obsolete APIs and docs.

New Features:
- Add a fallible LengthFormat::try_new constructor with input validation and expose bytes() and endianness() accessors.
- Introduce a new_length_codec method on connections to consistently apply LengthDelimitedCodec capped by buffer_capacity.
- Add tests to verify honouring of buffer_capacity in send_response and process_stream, and to reject invalid run_app capacities.

Bug Fixes:
- Validate length-prefix width at runtime and panic or error on invalid sizes to prevent misuse.
- Fix tests to assert empty buffers after decoding to catch unexpected trailing data.

Enhancements:
- Replace all custom framing code (FrameProcessor and LengthPrefixedProcessor) with tokio_util::codec::LengthDelimitedCodec across tests, helpers, examples, builder, and connection logic.
- Unify framing setup via a new_test_codec helper and TEST_MAX_FRAME constant in wireframe_testing.
- Ensure send_response and stream processing consume only the length prefix on partial frames and assert no trailing bytes remain.

Documentation:
- Revise all design documents, README, and examples to document codec-based framing instead of the old processor APIs and improve formatting and terminology.
- Remove or rephrase obsolete sections on FrameProcessor in docs.

Tests:
- Update existing tests to use LengthDelimitedCodec and new_test_codec helper.
- Add helper tests for run_app rejecting zero and excessive capacities.

Chores:
- Remove the src/frame/processor.rs module, builder.frame_processor field, SharedState::new, efficiency-report, and other deprecated framing artifacts.
- Clean up builder.rs and remove deprecated frame_processor methods.

### CODE REVIEW

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix:

Inbound processing now builds a LengthDelimitedCodec with default settings. That codec caps frames at 8 MiB by default, whereas the builder still exposes buffer_capacity up to 16 MiB and the previous LengthPrefixedProcessor accepted any usize length. After this change, frames between 8 MiB and the configured capacity will be rejected and the connection closed even though the configuration suggests they are allowed; the same default is used in send_response, so large responses fail to encode as well. The codec should be initialised with an explicit max_frame_length (and mirrored on the send path) derived from the configured buffer size to avoid the regression.

**Comment by leynos** [comment]
@coderabbitai Has the following now been resolved?

Inbound processing now builds a LengthDelimitedCodec with default settings. That codec caps frames at 8 MiB by default, whereas the builder still exposes buffer_capacity up to 16 MiB and the previous LengthPrefixedProcessor accepted any usize length. After this change, frames between 8 MiB and the configured capacity will be rejected and the connection closed even though the configuration suggests they are allowed; the same default is used in send_response, so large responses fail to encode as well. The codec should be initialised with an explicit max_frame_length (and mirrored on the send path) derived from the configured buffer size to avoid the regression.

**Comment by leynos** [comment]
@coderabbitai re:

> 56-58: Use en‑GB spelling in prose.

> Write “deserialises” in documentation prose; keep US spelling only for API names.

> -then waits for a frame, decodes it, and
> -deserializes the response type.
> +then waits for a frame, decodes it, and
> +deserialises the response type.

"deserializes" is correct for en-GB-oxendict (-ize, -yse, -our) as used by this project

File: src/frame/mod.rs [hunk 1]
```diff
@@ -1,12 +1,12 @@
-//! Frame encoding utilities and length-prefixed processors.
+//! Frame encoding utilities and length-prefix helpers.
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/frame/mod.rs]: @codex has this issue been resolved? Please provide evidence for your assessment.

If work remains, please provide clear self-contained instructions (including rationale and location) in a triple-tilde code-fenced block.

File: docs/asynchronous-outbound-messaging-design.md [hunk 2]
```diff
@@ -707,10 +707,10 @@ features of the 1.0 release.
...
 
 - **Message Fragmentation:** Pushes occur at the *logical frame* level. The
-  `FragmentAdapter` will operate at a lower layer in the `FrameProcessor`
-  stack, transparently splitting any large pushed frames before they are
-  written to the socket. The `PushHandle` and the application code that uses it
-  remain completely unaware of fragmentation.
+  `FragmentAdapter` will operate at a lower layer in the codec stack,
+  transparently splitting any large pushed frames before they are written to
+  the socket. The `PushHandle` and the application code that uses it remain
+  completely unaware of fragmentation.
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/asynchronous-outbound-messaging-design.md:744]: @codex has this issue been resolved? Please provide evidence for your assessment.

If work remains, please provide clear self-contained instructions (including rationale and location) in a triple-tilde code-fenced block.

File: README.md [hunk 3]
```diff
@@ -126,16 +126,15 @@ Use `None` rather than `Some(0)` when a frame lacks a correlation ID. See
...
+encoded using the application's configured serializer and framed by a
+length‑delimited codec【F:docs/rust-binary-router-library-design.md†L724-L730】.
 
-The included `LengthPrefixedProcessor` illustrates a simple framing strategy
-that prefixes each frame with its length. The format is configurable (prefix
-size and endianness) and defaults to a 4‑byte big‑endian length
-prefix【F:docs/rust-binary-router-library-design.md†L1082-L1123】.
+Frames are length prefixed using `tokio_util::codec::LengthDelimitedCodec`. The
+prefix length and byte order are configurable and default to a 4‑byte
+big‑endian header【F:docs/rust-binary-router-library-design.md†L1082-L1123】.
```
[the remark this answers is not in the document]
Comment (leynos) [reply README.md:137]: @codex has this issue been resolved? Please provide evidence for your assessment.

If work remains, please provide clear self-contained instructions (including rationale and location) in a triple-tilde code-fenced block.

File: src/app/connection.rs [hunk 4]
```diff
@@ -55,7 +55,9 @@ where
             .serializer
             .serialize(msg)
             .map_err(SendError::Serialize)?;
-        let mut codec = LengthDelimitedCodec::builder().new_codec();
+        let mut codec = LengthDelimitedCodec::builder()
+            .max_frame_length(self.buffer_capacity)
+            .new_codec();
         let mut framed = BytesMut::new();
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/app/connection.rs:61]: @codex has this issue been resolved? Please provide evidence for your assessment.

If work remains, please provide clear self-contained instructions (including rationale and location) in a triple-tilde code-fenced block.

File: docs/generic-message-fragmentation-and-re-assembly-design.md [hunk 5]
```diff
@@ -156,19 +157,44 @@ pub trait FragmentStrategy: 'static + Send + Sync {
...
-       .route(...)
+        .codec(
+            LengthDelimitedCodec::builder()
+                .max_frame_length(64 * 1024)
+                .new_codec(),
+        )
+        .codec(FragmentAdapter::new(MySqlStrategy::new()))
+        .route(/* ... */)
 })
 ```
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/generic-message-fragmentation-and-re-assembly-design.md]: @codex has this issue been resolved? Please provide evidence for your assessment.

If work remains, please provide clear self-contained instructions (including rationale and location) in a triple-tilde code-fenced block.

File: docs/wireframe-client-design.md [hunk 6]
```diff
@@ -54,15 +58,15 @@
 
 Internally, this uses the `Serializer` to encode the request, sends it through
 the length‑delimited codec, then waits for a frame, decodes it, and
-deserializes the response type.
+deserialises the response type.
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/wireframe-client-design.md]: @codex has this issue been resolved? Please provide evidence for your assessment.

If work remains, please provide clear self-contained instructions (including rationale and location) in a triple-tilde code-fenced block.

File: docs/wireframe-client-design.md [hunk 7]
```diff
@@ -54,15 +58,15 @@
...
 
-### Connection Lifecycle
+### Connection lifecycle
 
 Like the server, the client should expose hooks for connection setup and
-teardown. These mirror the server’s lifecycle callbacks so both sides can share
-initialization logic.
+teardown. These mirror the server’s lifecycle callbacks, so both sides can
+share initialisation logic.
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/wireframe-client-design.md]: @codex has this issue been resolved? Please provide evidence for your assessment.

If work remains, please provide clear self-contained instructions (including rationale and location) in a triple-tilde code-fenced block.

File: README.md [hunk 8]
```diff
@@ -126,22 +126,21 @@ Use `None` rather than `Some(0)` when a frame lacks a correlation ID. See
...
```
[the remark this answers is not in the document]
Comment (leynos) [reply README.md]: @codex has this issue been resolved? Please provide evidence for your assessment.

If work remains, please provide clear self-contained instructions (including rationale and location) in a triple-tilde code-fenced block.

File: README.md [hunk 9]
```diff
@@ -283,7 +282,7 @@
 Example programs are available in the `examples/` directory:
 
 - `echo.rs` — minimal echo server using routing
-- `ping_pong.rs` — showcases serialization and middleware in a ping/pong
+- `ping_pong.rs` — showcases serialisation and middleware in a ping/pong
   protocol. See [examples/ping_pong.md](examples/ping_pong.md) for a detailed
```
[the remark this answers is not in the document]
Comment (leynos) [reply README.md]: @codex has this issue been resolved? Please provide evidence for your assessment.

If work remains, please provide clear self-contained instructions (including rationale and location) in a triple-tilde code-fenced block.

File: tests/routes.rs [hunk 10]
```diff
@@ -9,17 +9,19 @@ use std::sync::{
...
-    frame::{FrameProcessor, LengthPrefixedProcessor},
     message::Message,
     serializer::BincodeSerializer,
 };
 use wireframe_testing::{drive_with_bincode, drive_with_frames};
 
 type TestApp = wireframe::app::WireframeApp<BincodeSerializer, (), TestEnvelope>;
 
+const MAX_FRAME: usize = 64 * 1024;
+
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/routes.rs]: @codex has this issue been resolved? Please provide evidence for your assessment.

If work remains, please provide clear self-contained instructions (including rationale and location) in a triple-tilde code-fenced block.

File: wireframe_testing/src/helpers.rs [hunk 11]
```diff
@@ -353,8 +347,11 @@ where
             format!("bincode encode failed: {e}"),
         )
     })?;
+    let mut codec = LengthDelimitedCodec::builder()
+        .max_frame_length(DEFAULT_CAPACITY)
+        .new_codec();
     let mut framed = BytesMut::new();
-    LengthPrefixedProcessor::default().encode(&bytes, &mut framed)?;
+    codec.encode(bytes.into(), &mut framed)?;
     drive_with_frame(app, framed.to_vec()).await
```
[the remark this answers is not in the document]
Comment (leynos) [reply wireframe_testing/src/helpers.rs]: @codex has this issue been resolved? Please provide evidence for your assessment.

If work remains, please provide clear self-contained instructions (including rationale and location) in a triple-tilde code-fenced block.

File: src/app/connection.rs [hunk 12]
```diff
@@ -158,7 +164,7 @@ where
     where
         W: AsyncRead + AsyncWrite + Unpin,
     {
-        let codec = LengthDelimitedCodec::builder().new_codec();
+        let codec = self.new_length_codec();
         let mut framed = Framed::new(stream, codec);
         framed.read_buffer_mut().reserve(self.buffer_capacity);
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/app/connection.rs:171]: @codex has this issue been resolved? Please provide evidence for your assessment.

If work remains, please provide clear self-contained instructions (including rationale and location) in a triple-tilde code-fenced block.
